CONTRIBUTING.md

```diff
@@ -258,3 +258,18 @@ the main branch. The workflow file should be named as
 [![NuGet version badge](https://img.shields.io/nuget/v/{your_package_name})](https://www.nuget.org/packages/{your_package_name})
 [![NuGet download count badge](https://img.shields.io/nuget/dt/{your_package_name})](https://www.nuget.org/packages/{your_package_name})
 ```
+
+### Guidance for target frameworks
+
+Components shipped from this repo SHOULD comply with the following requirements:
+
+* SHOULD support the [.NET
+  versions](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main#supported-net-versions)
+  which are supported by the OpenTelemetry main repo packages.
+* SHOULD not support out-of-support .NET runtimes (eg: `.NET 452`, `.NET Core
+  2.1` etc). CI checks in this repository will not be run against out of support
```

Comment:
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1031#issuecomment-1450622575 Are we allowing/disallowing a component to define own CI ?

Reply:
Disallowing to define own CI for supporting out of support runtimes. At present we do not have any components that need this and for new components do not anticipate the need to support older versions.